q2_demux/_subsample.py

```diff
@@ -0,0 +1,95 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import gzip
+import random
+
+import pandas as pd
+
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt,
+    FastqManifestFormat)
+
+from q2_demux._demux import _read_fastq_seqs, _write_metadata_yaml
+
+
+def subsample_single(sequences: SingleLanePerSampleSingleEndFastqDirFmt,
+                     fraction: float
+                     ) -> SingleLanePerSampleSingleEndFastqDirFmt:
+    result = SingleLanePerSampleSingleEndFastqDirFmt()
+
+    manifest_path = os.path.join(str(sequences), sequences.manifest.pathspec)
+
+    manifest = pd.read_csv(manifest_path, header=0, comment='#')
+
+    for fwd_name in manifest.filename.tolist():
```

Comment:
```suggestion
    for _, fwd_name in df.itertuples():
```

This suggestion is effectively the same as what you had originally, but, it translates to simpler code in the paired-end code, so I propose updating here for consistency.